### PR TITLE
Update and re-enable Sphinx documentation

### PR DESCRIPTION
**Pull request by sethrj**
- Re-add missing "make doc"
- Restructure documentation (fewer nested levels)
- Update documentation for independent ForTrilinos build
- Fix RTD support (@aprokop is it possible to add me as admin for https://readthedocs.org/projects/fortrilinos/ )?

### CODE REVIEW

**Comment by aprokop** [comment]
>     * @aprokop is it possible to add me as admin for [readthedocs.org/projects/fortrilinos](https://readthedocs.org/projects/fortrilinos/)

Done.

File: doc/overview.rst [hunk 1]
```diff
@@ -24,10 +27,10 @@ ForTrilinos is developed and maintained by:
 
 * `Seth Johnson <sethrj@ornl.gov>`_
```
Comment (aprokop) [inline doc/overview.rst:28]: Make yourself first :)

File: doc/conf.py [hunk 2]
```diff
@@ -49,17 +49,22 @@
 
 # General information about the project.
 project = 'ForTrilinos'
-copyright = '2017, Andrey Prokopenko, Seth Johnson'
-author = 'Andrey Prokopenko, Seth Johnson'
+copyright = '2020, Oak Ridge National Laboratory, UT-Battelle, LLC'
```
Comment (aprokop) [inline doc/conf.py:52]: 2017-2020?

**Review by aprokop — COMMENTED** [review]
Looks fine. You no longer need Docker info?

File: doc/overview.rst (hunk 1 above)
Comment (sethrj) [reply doc/overview.rst:28]: And corrected my email too 😆

**Comment by sethrj** [comment]
Regarding docker: I thought it easier to delete the docker information for now since it's out of date. Do you think I should try to update it?

**Comment by aprokop** [comment]
> Do you think I should try to update it?

Maybe, but it's low priority.

File: doc/conf.py [hunk 3]
```diff
@@ -49,7 +49,7 @@
 
 # General information about the project.
 project = 'ForTrilinos'
-copyright = '2020, Oak Ridge National Laboratory, UT-Battelle, LLC'
+copyright = '2017­2020, Oak Ridge National Laboratory, UT-Battelle, LLC'
```
Comment (aprokop) [inline doc/conf.py:52]: :(

Comment (sethrj) [reply doc/conf.py:52]: 🤦

Comment (sethrj) [reply doc/conf.py:52]: wait, this looks like a unicode error with the en dash:
```
-copyright = '2020, Oak Ridge National Laboratory, UT-Battelle, LLC'
+copyright = '2017<U+00AD>2020, Oak Ridge National Laboratory, UT-Battelle, LLC'
```
guess I have to read up on github unicode... I thought the 
```
# -*- coding: utf-8 -*-
```
at the beginning was supposed to take care of that.

Comment (sethrj) [reply doc/conf.py]: OK, looks like my version of vim inserted the wrong unicode character even though it displays correctly inline and shows the right `fileencoding`? 😩 Should be good now.

**Review by aprokop — APPROVED** [review]

**Comment by aprokop** [comment]
Need to rebase on master.

(and `Fix uniode character`? You are on a roll. Please not fix that :))

**Pull request merged**